File: benchmarks/src/ruby/closure_chain.rb
def make_counter(start)
    c = start
    return lambda { c = c + 1; c }
end

N = 150000
acc = 0
start = Time.now.to_f * 1000
N.times do |i|
    ctr = make_counter(i % 1000)
    acc = (acc + ctr.call + ctr.call + ctr.call) % 1000003
end
finish = Time.now.to_f * 1000
puts "Result: #{acc}"
puts "Time: #{(finish - start).round}ms"

File: benchmarks/src/nodejs/closure_chain.js
function makeCounter(start) {
    let c = start;
    return function() {
        c = c + 1;
        return c;
    };
}

const N = 150000;
let acc = 0;
const start = Date.now();
for (let i = 0; i < N; i++) {
    const ctr = makeCounter(i % 1000);
    acc = (acc + ctr() + ctr() + ctr()) % 1000003;
}
const end = Date.now();
console.log("Result: " + acc);
console.log("Time: " + Math.floor(end - start) + "ms");
